### ISSUE TITLE
Balance Configuration for Objectness Loss Calculation

### ISSUE THREAD

**Issue by Zailux** [eriklindernoren/PyTorch-YOLOv3#681]
## What I'm trying to do
Hi everyone,

I am trying to use this project to train a 5 Layer YOLO v3 model, which recommend in the Darknet Wiki if I want to recognize small and large objects with YOLO. 
See Config https://raw.githubusercontent.com/AlexeyAB/darknet/master/cfg/yolov3_5l.cfg

I am quite a beginner regarding deep learning , but I what I am trying to understand is the balancing of the objectness loss, in the compute_loss method. 
I just see, that you've remove these "magic values" in a recent commit. Can anybody nonetheless explain me, what the thought behind these values were? 
https://github.com/eriklindernoren/PyTorch-YOLOv3/commit/d84e7cd90cdc5b67c82e7f8d0ebe7d1294242536

```
#Line 156
balance = [4.0, 1.0, 0.4, 0.1]
# line 188
lobj += BCEobj(layer_predictions[..., 4], tobj) * balance[layer_index] # obj loss
```

I had to extend these values without understanding them, in my fork because they'd throw me an error, because this config only covers 4 YOLO layers and not 5. 
Thanks a lot. 

Kind regards
Zailux :)

**Comment by Flova** [comment]
Hey,

I think the original idea was to balance the loss between the diffentent output layers to help with convergence and balance the contribution of the different layers. The values where found empirical. The values where removed because they might only work well for coco and introduce complexity with a small benefit.

Cheers
